nav2_behavior_tree/include/nav2_behavior_tree/bt_action_server_impl.hpp

```diff
@@ -67,16 +68,22 @@ BtActionServer<ActionT>::BtActionServer(
   };
 
   if (!node->has_parameter("error_code_names")) {
-    std::string error_codes_str;
-    for (const auto & error_code : error_code_names) {
-      error_codes_str += error_code + "\n";
-    }
-    RCLCPP_WARN_STREAM(
-      logger_, "Error_code parameters were not set. Using default values of: "
-        << error_codes_str
-        << "Make sure these match your BT and there are not other sources of error codes you want "
-        "reported to your application");
+
+    const rclcpp::ParameterValue value = node->declare_parameter(
+      "error_code_names", 
+      rclcpp::PARAMETER_STRING_ARRAY);
+    if (value.get_type() == rclcpp::PARAMETER_NOT_SET) {
+      std::string error_codes_str;
+      for (const auto & error_code : error_code_names) {
+        error_codes_str += error_code + "\n";
+      }
+      RCLCPP_WARN_STREAM(
+          logger_, "Error_code parameters were not set. Using default values of: "
+          << error_codes_str
+          << "Make sure these match your BT and there are not other sources of error codes you"
+          "reported to your application");
     node->declare_parameter("error_code_names", error_code_names);
```

Comment:
This redeclared the parameter that was already declared... 

I think it should be setting

Reply:
Its indentation also looks wrong, so not sure if its also an issue of being in the wrong conditional

Reply:
Maybe this way, without redeclaration?

```
if (!node->has_parameter("error_code_names")) {
  node->declare_parameter(
    "error_code_names",
    rclcpp::PARAMETER_STRING_ARRAY);
}

rclcpp::Parameter tmp;
if (!node->get_parameter("error_code_names", tmp)) {
  node->set_parameter(rclcpp::Parameter("error_code_names", error_code_names));
  
  std::vector<std::string> error_code_names = {
  "follow_path_error_code",
  "compute_path_error_code"
  };

  std::string error_codes_str;
  for (const auto & error_code : error_code_names) {
    error_codes_str += error_code + "\n";
  }
  RCLCPP_WARN_STREAM(
    logger_, "Error_code parameters were not set. Using default values of: "
      << error_codes_str
      << "Make sure these match your BT and there are not other sources of error codes you want "
      "reported to your application");
}
```

Reply:
That wouldn't compile for a few reasons (e.g. setting error code names before its defined) but yes, that's generally speaking what I think would work well.